lib/govuk_index/popularity_updater.rb

```diff
@@ -0,0 +1,62 @@
+module GovukIndex
+  class PopularityUpdater
+    SCROLL_BATCH_SIZE = 500
+    PROCESSOR_BATCH_SIZE = 25
+    TIMEOUT_SECONDS = 30
+
+    def self.update(index_name)
+      new(
+        source_index: index_name,
+        destination_index: index_name,
+      ).run
+    end
+
+    # This task is designed to migrate data when the schema changes, It waits for the
+    # queued jobs to be processed, this is done by monitoring the sidekiq queue and workers.
+    # This process should be closely monitored as it is not gauranteed to work.
+    def self.migrate(index_name)
+      raise('Please set `CONFIRM_INDEX_MIGRATION_START` to continue') unless ENV['CONFIRM_INDEX_MIGRATION_START']
+
+      index_group = SearchConfig.instance.search_server.index_group(index_name)
+      new_index = index_group.create_index
+      index_group.current.with_lock do
+        new(
+          source_index: index_group.current.real_name,
+          destination_index: new_index.real_name,
+        ).run
+        # wait for queued tasks to be started
+        sleep 1 while(Sidekiq::Queue.new(GovukIndex::PopularityWorker::QUEUE_NAME).size > 0) # rubocop: disable Style/ZeroLengthPredicate
+        # wait for started tasks to be finished
```

Comment:
I'm probably misreading the code, but it looks like it would do this:

1. Create a new index, but don't switch to it
2. Lock the current, pre-existing index
3. Wait for queued tasks to insert data into the current index (which is locked, so I'd expect insertions to fail)

Have I misunderstood step one of these steps? What causes the data to be inserted into the new index when the "current" index is still the old one?